Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt

```diff
@@ -105,55 +87,42 @@ open class BpkText @JvmOverloads constructor(
 
   private fun initialize(attrs: AttributeSet?, defStyleAttr: Int) {
 
-    val styleAttr = when (defStyleAttr) {
-      0 -> R.attr.bpkTextStyle
-      else -> defStyleAttr
-    }
-
-    val a: TypedArray = context.theme.obtainStyledAttributes(
+    context.theme.obtainStyledAttributes(
       attrs,
       R.styleable.BpkText,
-      styleAttr, 0)
-
-    textStyle = a.getInt(R.styleable.BpkText_textStyle, BASE)
-    val weightArg = a.getInt(R.styleable.BpkText_weight, -1)
-    if (weightArg == -1) {
-      // if weight has not been set we still read the emphasize property
-      emphasize = a.getBoolean(R.styleable.BpkText_emphasize, false)
-    } else {
-      weight = Weight.values()[weightArg]
-    }
+      defStyleAttr, 0
+    ).use {
+      _textStyle = it.getInt(R.styleable.BpkText_textStyle, BASE)
```

Comment:
Let's use a local variable instead of backing property. We can do something like:
```
val property = BASE
...use {
propery = it.getInt(styleable, property)
}
this.property = property;
```

Reply:
This will trigger the `setup` function twice in the constructor, that's why I moved it